### PR TITLE
#175494040 Configuration of a sequelize 

### PR DESCRIPTION
**Pull request by LouangeMu**
### What does this PR do?

- setting up sequelize.

### Description of PR to be completed

- This sequelize is going to be used as our ORM on the API. 
- Sets up the migrations and model structure, have a dummy user table and model setup to test the whole configuration. 

### How should this be manually tested

  -  Clone the project `https://github.com/Soma-Technologies-Inc/visitAfrica-backend.git`
  -   `cd visitAfrica-backend`
  -    Install dependencies `npm i`
  -    `git fetch`  to download the latest commits
  -    To switch to this branch `ch-config-sequelize-175494040`
  
 ### Any background context you want to provide?
  N/A

### What are the relevant pivotal tracker stories?

[#175494040](https://www.pivotaltracker.com/story/show/175494040)

### CODE REVIEW

File: src/database/config/config.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<<<<<<< HEAD
```
Comment (tigthor) [inline src/database/config/config.js]: remove this conflict

File: src/database/config/config.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import {config} from 'dotenv';
+config();
+
+export const development = {
+=======
+require('dotenv').config();
+
+module.exports.development = {
+>>>>>>> dfe129ac0e00169f2ec4039c2720730a4bf20050
```
Comment (tigthor) [inline src/database/config/config.js]: These conflicts should be removed

File: src/database/models/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+});
+
+db.sequelize = sequelize;
+db.Sequelize = Sequelize;
+
+<<<<<<< HEAD
+export default db;
+=======
+module.exports = db;
+>>>>>>> dfe129ac0e00169f2ec4039c2720730a4bf20050
```
Comment (tigthor) [inline src/database/models/index.js]: still these are conflicts

**Comment by tigthor** [comment]
Ahhh All is fine

**Review by tigthor — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 package-lock.json
+.env
```
Comment (erickyvand) [inline .gitignore]: add new line

File: src/database/config/config.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require('dotenv').config();
```
Comment (erickyvand) [inline src/database/config/config.js]: import using `ES6 syntax`

File: src/database/models/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+const basename = _basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.js')[env];
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
```
Comment (erickyvand) [inline src/database/models/index.js]: This line of code `config.database, config.username, config.password,` should be replaced with `config.url`

**Review by erickyvand — CHANGES_REQUESTED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

**Pull request merged**